src/extensions/averagewaveforms/AverageWaveformsView/AverageWaveformView.tsx

```diff
@@ -31,7 +31,7 @@ const AverageWaveformView: FunctionComponent<Props> = ({ sorting, recording, uni
         {
             sorting_object: sorting.sortingObject,
             recording_object: recording.recordingObject,
-            unit_id: unitId
+            unit_id: applyMergesToUnit(unitId, sorting.curation, selection.applyMerges)
```

Comment:
applyMergesToUnit will return either a single number of a list of numbers, depending on the curation, and whether user has selected to 'apply merges'